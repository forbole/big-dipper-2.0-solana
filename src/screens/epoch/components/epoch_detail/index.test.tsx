/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import EpochDetail from '.';

// ==================================
// mocks
// ==================================
jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
}));

jest.mock('./components', () => ({
  Mobile: (props) => <div id="Mobile" {...props} />,
  Desktop: (props) => <div id="Desktop" {...props} />,
}));

jest.mock('@material-ui/core', () => ({
    Typography: (props) => <div id="Typography" {...props} />,
}));


// ==================================
// unit tests
// ==================================
describe('screen: Epoch/EpochDetail', () => {
  it('matches snapshot', async () => {
    let component;
    renderer.act(() => {
      component = renderer.create(
        <MockTheme>
          <EpochDetail
            title = "inflationGovernor"
            epochDetail = {[
                {
                    name: "Foundation",
                    value: 0,
                    detail: "Percentage of total inflation allocated to the foundation",
                },
                {
                    name: "FoundationTerm",
                    value: 0,
                    detail: "Duration of foundation pool inflation in years",
                },
                {
                    name: "Initial",
                    value: 0.08,
                    detail: "The initial inflation percentage from time 0",
                },
                {
                    name: "tTaper",
                    value: 0.15,
                    detail: "Rate per year at which inflation is lowered. Rate reduction is derived using the target slot time in genesis config",
                },
                {
                    name: "Terminal",
                    value: 0.015,
                    detail: "Terminal inflation percentage",
                },
            ]}
          />
        </MockTheme>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});

