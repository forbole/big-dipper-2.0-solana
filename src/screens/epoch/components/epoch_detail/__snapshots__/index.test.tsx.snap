// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Epoch/EpochDetail matches snapshot 1`] = `
<div
  className=""
  id="Box"
>
  <div
    id="Typography"
    variant="h2"
  >
    inflationGovernor
  </div>
</div>
`;
